sklearn/decomposition/_nmf.py

```diff
@@ -775,11 +807,26 @@ def _fit_multiplicative_update(X, W, H, beta_loss='frobenius',
 
     References
     ----------
+    Lee, D. D., & Seung, H., S. (2001). Algorithms for Non-negative Matrix
+    Factorization. Adv. Neural Inform. Process. Syst.. 13.
     Fevotte, C., & Idier, J. (2011). Algorithms for nonnegative matrix
     factorization with the beta-divergence. Neural Computation, 23(9).
     """
     start_time = time.time()
 
+    n_samples = X.shape[0]
+    max_iter_update_h_ = 1
```

Comment:
You don't seem to use `max_iter_update_h_ > 1` anymore. This should be cleaned up.

Reply:
Yes, it wasn't clear to me if 'internal' iterations was still necessary: you don't seem to be shocked by their absence... :) so I have cleaned up all the relics.